test/functional/tests/incremental_load/test_udev.py

```diff
@@ -0,0 +1,235 @@
+        description: |
+          Verify if CAS udev rule is executed for core when its cache instance is not loaded
+          and loaded.
+        pass_criteria:
+          - No kernel error
+          - Core devices are listed in core pool when cache is not available
+          - Core devices are moved from core pool and attached to cache after plugging cache device
+    """
+    with TestRun.step("Start cache and add core."):
+        cache_disk = TestRun.disks["cache"]
+        cache_disk.create_partitions([Size(1, Unit.GibiByte)])
+        cache_dev = cache_disk.partitions[0]
+        core_disk = TestRun.disks["core"]
+        core_disk.create_partitions([Size(2, Unit.GibiByte)])
+        core_dev = core_disk.partitions[0]
+        cache = casadm.start_cache(cache_dev, force=True)
+        core = cache.add_core(core_dev)
+    with TestRun.step("Create init config from running CAS configuration."):
+        InitConfig.create_init_config_from_running_configuration()
+    with TestRun.step("Stop cache."):
+        cache.stop()
+    with TestRun.step("Unplug core device."):
+        core_disk.unplug()
+    with TestRun.step("Plug core device."):
+        core_disk.plug()
+    with TestRun.step("Check if core device is listed in core pool."):
+        check_if_dev_in_core_pool(core_dev)
+    with TestRun.step("Unplug cache device."):
+        cache_disk.unplug()
+    with TestRun.step("Plug cache disk."):
+        cache_disk.plug()
+    with TestRun.step("Check if core device is active and not in the core pool."):
+        check_if_dev_in_core_pool(core_dev, False)
+        if core.get_status() != CoreStatus.active:
+            TestRun.fail(f"Core status is {core.get_status()} instead of active.")
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.parametrize("cache_mode", CacheMode)
+def test_udev_cache_load(cache_mode):
+    """
+        title: CAS udev rule execution after unplugging and plugging cache device.
+        description: |
+          Verify if CAS udev rule is executed after unplugging and plugging cache device and if
+          cache is properly loaded.
+        pass_criteria:
+          - No kernel error
+          - Cache is properly loaded after plugging cache device
+          - Cache is not loaded after unplugging cache device
+    """
+    with TestRun.step("Start cache."):
+        cache_disk = TestRun.disks["cache"]
+        cache_disk.create_partitions([Size(1, Unit.GibiByte)])
+        cache_dev = cache_disk.partitions[0]
+        cache = casadm.start_cache(cache_dev, cache_mode=cache_mode)
+    with TestRun.step("Create init config from running configuration"):
+        InitConfig.create_init_config_from_running_configuration()
+    with TestRun.step("Stop cache."):
+        cache.stop()
+    with TestRun.step("Unplug cache disk."):
+        cache_disk.unplug()
+    with TestRun.step("List caches and check if cache is not listed."):
```

Comment:
Is this step necessary? Cache is being stopped in line 120 and I cannot imagine a possible scenario where stopped cache would reappear in casadm -L after unplugging cache device.